app/src/main/java/org/wikipedia/analytics/metricsplatform/ArticleEvent.kt

```diff
@@ -219,10 +227,15 @@ class ArticleLinkPreviewInteraction : TimedMetricsEvent {
     private fun submitEvent(action: String) {
         submitEvent(
             "article_link_preview_interaction",
-            mapOf(
-                "action" to action,
-                "source" to source,
-                "time_spent_ms" to timer.elapsedMillis,
+            getInteractionData(
+                "article_link_preview_interaction",
+                action,
+                source.toString(),
+                "time_spent_ms.$timer.elapsedMillis",
+                null,
+                null,
+                null,
+                null
```

Comment:
Same as above:

```
getInteractionData(
    "article_link_preview_interaction",
    action,
    source.toString(),
    "time_spent_ms.$timer.elapsedMillis"
)
```

Reply:
Also another minor change would be to change it to `"time_spent_ms.${timer.elapsedMillis}"` to get the ellapsedMillis value, else it will just give a string representation of the timer object.
Also curious, is this the variable where we can compose and add a string with extra data that might need to go with the event?

Reply:
thanks @cooltey ! removed extraneous nulls!

thanks @sharvaniharan ! updated the timer value

> is this the variable where we can compose and add a string with extra data that might need to go with the event?

Yes - [InteractionData](https://gitlab.wikimedia.org/repos/data-engineering/metrics-platform/-/blob/main/java/src/main/java/org/wikimedia/metrics_platform/context/InteractionData.java?ref_type=heads) maps to the [MP common fragment](https://gerrit.wikimedia.org/r/plugins/gitiles/schemas/event/secondary/+/refs/heads/master/jsonschema/fragment/analytics/product_metrics/common/current.yaml) which is referenced from the [MP base app schema](https://gerrit.wikimedia.org/r/plugins/gitiles/schemas/event/secondary/+/refs/heads/master/jsonschema/analytics/product_metrics/app/base/current.yaml). As you can see, the properties of `InteractionData` are mostly strings so data can be appended to any or all of them where appropriate.